storage/series/series.go

```diff
@@ -440,6 +442,31 @@ func (s *dbSeries) mergeBlock(
 	return nil
 }
 
+func (s *dbSeries) Update(blk block.DatabaseBlock) error {
+	s.Lock()
+	defer s.Unlock()
+
+	// ensure series is bootstraped before allowing any updates
+	if s.bs != bootstrapped {
+		return errSeriesNotBootstrapped
+	}
+
+	// we only accept updates to non-buffered data
+	// ensure buffer is past provided time
+	min, _ := s.buffer.MinMax()
+	if !blk.StartTime().Before(min) {
```

Comment:
AFAIR `MinMax` returns the min and max timestamp of datapoints in the buffer, which means if `blk.StartTime()` is the start of the current block and the earliest datapoint in the buffer is later than that, we would still repair the block, which is probably not what you want.